pkg/querier/distributor_queryable.go

```diff
@@ -151,6 +169,21 @@ func (q *distributorQuerier) streamingSelect(ctx context.Context, minT, maxT int
 		sets = append(sets, series.NewConcreteSeriesSet(serieses))
 	}
 
+	streamingSeries := make([]storage.Series, 0, len(results.StreamingSeries))
+	for _, s := range results.StreamingSeries {
+		streamingSeries = append(streamingSeries, &streamingChunkSeries{
+			labels:            s.Labels,
+			chunkIteratorFunc: q.chunkIterFn,
+			mint:              minT,
+			maxt:              maxT,
```

Comment:
Minor: chunkIteratorFunc, mint, maxt don't change between series. I think one way to avoid having it store it in each `streamingChunkSeries` is to build our own `storage.SeriesSet` implementation which retains chunkIteratorFunc, mint, maxt and then implement `Iterator()` function which calls `GetChunks()` on `streamingChunkSeries`.

Reply:
In the interests of keeping this PR as small as possible (it's already enormous :)), I'm going to leave this as-is and come back to address this as a separate PR once this is merged.